packages/protocol/ledger/storage.go

```diff
@@ -258,32 +258,58 @@ func (s *Storage) initConsumers(outputIDs utxo.OutputIDs, txID utxo.TransactionI
 // pruneTransaction removes a Transaction (and all of its dependencies) from the database.
 func (s *Storage) pruneTransaction(txID utxo.TransactionID, pruneFutureCone bool) {
 	for futureConeWalker := walker.New[utxo.TransactionID]().Push(txID); futureConeWalker.HasNext(); {
+		spentOutputsWithMetadata := make([]*OutputWithMetadata, 0)
+
 		currentTxID := futureConeWalker.Next()
 
 		s.CachedTransactionMetadata(currentTxID).Consume(func(txMetadata *TransactionMetadata) {
 			s.CachedTransaction(currentTxID).Consume(func(tx utxo.Transaction) {
 				for it := s.ledger.Utils.ResolveInputs(tx.Inputs()).Iterator(); it.HasNext(); {
-					s.consumerStorage.Delete(byteutils.ConcatBytes(lo.PanicOnErr(it.Next().Bytes()), lo.PanicOnErr(currentTxID.Bytes())))
-				}
+					inputID := it.Next()
+
+					s.CachedOutput(inputID).Consume(func(output utxo.Output) {
```

Comment:
If we want to keep this method there is not need to collect stuff anymore to trigger the orphaned event, but just delete the inputs.